src/sourceFetch.js

```diff
@@ -0,0 +1,30 @@
+import ErrorPopupHandler from "./errorHandler.js";
+
+export default class Api {
+  constructor(apiUrl, apiKey, sourceID) {
+    this.apiUrl = apiUrl;
+    this.apiKey = apiKey;
+    this.data = "";
+    this.sourceID = sourceID;
```

Comment:
Why do you need sourceId here? It is dynamic value and you can put it as paramter in get request?